src/serializer/ResidualHeapVisitor.js

```diff
@@ -478,6 +491,11 @@ export class ResidualHeapVisitor {
       if (this._mark(equivalentValue)) this.visitAbstractValue(equivalentValue);
       return (equivalentValue: any);
     }
+    if (val instanceof ObjectValue && isReactElement(val)) {
+      let equivalentReactElementValue = this.reactElementEquivalenceSet.add(val);
+      if (this._mark(equivalentReactElementValue)) this.visitValueObject(equivalentReactElementValue);
+      return (equivalentReactElementValue: any);
```

Comment:
Prefer more specific cast if possible.

Reply:
I tried to, but battled endlessly with Flow issues because the function has a generic used. The generic in this case should be fine as `Value` though. Unless you can think of a way of making it work? Casting to `ObjectValue` seems to break the world.

Reply:
Hmm, maybe that's because you're accessing things specific to React elements and there should be a ReactElementValue type in the future that is a subtype of object or something.